edge/yaml/docker_compose_test.go

```diff
@@ -1,109 +1,84 @@
+	if err != nil {
+		t.Errorf("error while updating service with environment variables: %s", err)
+	}
+
+	// Verify that the YAML contains the added environment variables
+	if !strings.Contains(updatedYAML, "ENV_VAR_1=value1") || !strings.Contains(updatedYAML, "ENV_VAR_2=value2") {
+		t.Errorf("expected environment variables not found in the updated YAML: %s", updatedYAML)
+	}
+}
+
+func TestExtractRegistryServerUrl(t *testing.T) {
 	tests := []struct {
-		name     string
-		input    string
-		expected string
-		wantErr  bool
+		name      string
+		imageName string
+		expected  string
+		err       error
 	}{
 		{
-			name: "environment stanza does not exist",
-			input: `version: "3"
-services:
-  updater:
-    image: portainer/portainer-updater:latest
-    labels:
-      - io.portainer.hideStack=true
-      - io.portainer.updater=true
-    command: ["portainer", "--image", "portainerci/portainer:2.18", "--env-type", "standalone"]
-    volumes:
-      - /var/run/docker.sock:/var/run/docker.sock`,
-			expected: `version: "3"
-services:
-  updater:
-    image: portainer/portainer-updater:latest
-    labels:
-      - io.portainer.hideStack=true
-      - io.portainer.updater=true
-    command:
-      - portainer
-      - --image
-      - portainerci/portainer:2.18
-      - --env-type
-      - standalone
-    environment:
-      - TEST_VAR_1=test_value_1
-      - TEST_VAR_2=test_value_2
-    volumes:
-      - /var/run/docker.sock:/var/run/docker.sock
-`,
-			wantErr: false,
+			name:      "input without registry server url",
+			imageName: "portainer/agent:latest",
+			expected:  "",
+			err:       nil,
 		},
 		{
-			name: "environment stanza exists",
-			input: `version: "3"
-services:
-  updater:
-    image: portainer/portainer-updater:latest
-    labels:
-      - io.portainer.hideStack=true
-      - io.portainer.updater=true
-    command: ["portainer", "--image", "portainerci/portainer:2.18", "--env-type", "standalone"]
-    environment:
-      - SKIP_PULL=1
-    volumes:
-      - /var/run/docker.sock:/var/run/docker.sock`,
-			expected: `version: "3"
-services:
-  updater:
-    image: portainer/portainer-updater:latest
-    labels:
-      - io.portainer.hideStack=true
-      - io.portainer.updater=true
-    command:
-      - portainer
-      - --image
-      - portainerci/portainer:2.18
-      - --env-type
-      - standalone
-    environment:
-      - SKIP_PULL=1
-      - TEST_VAR_1=test_value_1
-      - TEST_VAR_2=test_value_2
-    volumes:
-      - /var/run/docker.sock:/var/run/docker.sock
-`,
-			wantErr: false,
+			name:      "input with registry server url",
+			imageName: "example.com/portainer/agent:latest",
+			expected:  "example.com",
+			err:       nil,
 		},
 		{
-			name: "valid docker compose file content",
-			input: `version: "3"
-services:
-invalid_service:`,
-			expected: ``,
-			wantErr:  true,
+			name:      "empty image name",
+			imageName: "",
+			expected:  "",
+			err:       errors.New("No image name provided"),
```

Comment:
is something like `http://localhost:3000/image:tag` a possible registry? and what about `localhost:3000/image:tag`?

Reply:
just to be clear, I'm not sure if those are valid registries, but if they are, we need to take care of them

Reply:
I'm surprised to see how many valid image URL formats there are. To ensure we have adequate test coverage, I will be adding more unit test cases.

Reply:
maybe we can use https://github.com/distribution/distribution/blob/v2.7.1/reference/reference.go#L189 ?